cache/redis_cache.go

```diff
@@ -329,6 +341,22 @@ func (r *redisCache) Name() string {
 	return r.name
 }
 
+func (f *redisCache) checkAlive() {
+	for {
+		select {
+		case <-f.quitAliveCheck:
+			return
+		default:
+			f.alive = f.client.Ping(context.Background()).Err() == nil
+			time.Sleep(pingInterval)
```

Comment:
using:
```go
ticker := time.NewTicker(pingInterval)

  for {
      select {
      case <-done:
          return
      case t := <-ticker.C:
          ....
      }
  }

```

Will ensure that we ping at every ping interval, taking the time it takes to execute the Ping as well (e.g. if the ping interval is 5s and we take 1s to ping, the next tick will run after 4s).